addon/services/stack-provider.js

```diff
@@ -0,0 +1,22 @@
+import Service from '@ember/service';
+import { action } from '@ember/object';
+
+export default class StackProviderService extends Service {
```

Comment:
I might think about a more specific name for this service, since it's going to be registered into the host app's global service namespace; I could see someone seeing `stack-provider` in there and asking themselves _what the heck is that?_

Reply:
Agree. Also consider adding something in there about it being the stack provider for dialogs? `Dialog` is pretty generic and likely to cause a clash anyway (not much we can do about that), so how about `DialogStackProvider`?